u_fingerprinter.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2017 Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tls
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"golang.org/x/crypto/cryptobyte"
+)
+
+// Fingerprinter is a struct largely for holding options for the FingerprintClientHello func
+type Fingerprinter struct {
+	// KeepPSK will ensure that PSK extensions are passed along into the resulting ClientHelloSpec
+	KeepPSK bool
+	// AllowBluntMimicry will ensure that unknown extensions are passed along into the resulting ClientHelloSpec
+	// WARNING: there could be numerous subtle issues with ClientHelloSpecs that are generated with this flag
```

Comment:
Sufficient warning??

Reply:
I'd add that those issues may compromise security and mimicry